### ISSUE TITLE
Layout doesn't build with Xcode 9

### ISSUE THREAD

**Issue by nixterrimus** [schibsted/layout#38]
Building with Xcode 9 doesn't work unless Swift 3 is selected as the compiler

<img width="1419" alt="screen_shot_2017-08-27_at_11_58_23_am" src="https://user-images.githubusercontent.com/18454/29753127-7cdc2822-8b1f-11e7-94b1-26d299001b0a.png">

I realize Xcode 9 isn't out yet (and neither is the Swift 4 compiler) but I hope this issue helps someone who's wondering why their project isn't building.

**Comment by nicklockwood** [comment]
Yeah, I don't think Swift 4 support will be backwards compatible, so I'll have to maintain it as a separate branch until Swift 3.2 can be dropped.

**Comment by TMTBO** [comment]
```
post_install do |installer|
    layout = installer.pods_project.targets.find { |target| target.name == 'Layout' }
    layout.build_configurations.each do |config|
        config.build_settings['SWIFT_VERSION'] = '3.2'
    end
end
```

Add these at the end of your podfile, then `pod install` or `pod update`

**Comment by hartbit** [comment]
Instead of maintaining a separate branch, it might be better to use:

```
#if swift(>=4.0)
#else
#endif
```

Swift 4 is similar enough to Swift 3 that the changes should be minimal.

**Comment by nicklockwood** [comment]
@hartbit that makes sense.

**Comment by nicklockwood** [comment]
Fixed in 0.4.19
